docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+presented commands. Any Go version newer than 1.8 will do but it is considered a good
+practice to have the latest version of Go installed. You can check your Go version
+by executing `go version`.
+
+If you still need to install Go, you can follow our guide for Ubuntu installation [here](https://www.linode.com/docs/development/go/install-go-on-ubuntu/).
+
+{{< note >}}
+This guide is written for a non-root user. Depending on your configuration, some commands might require the help of `sudo` in order to get property executed. If you are not familiar with the `sudo` command, see the [Users and Groups](/docs/tools-reference/linux-users-and-groups/) guide.
+{{< /note >}}
+
+## About the context package
+
+The main purpose of the `context` package is to define the `Context` type and
+support *cancellation*. This happens because there are times where you want to
+abandon what you are doing but it an elegant way. However, it would be very helpful
+to be able to include some extra information about your cancelling decisions and propagate
+the information in a standard and portable way. The `context` package defines the
+`context.Context` type that can carry deadlines and cancellation signals between processes.
+
+If you take a look at the source code of the `context` package, you will realize that its
+implementation is pretty simple — even the implementation of the `Context` type is pretty
+simple, yet the `context` package is very important.
+
+The `Context` type is a Go *interface* with four methods, named `Deadline()`, `Done()`,
+`Err()`, and `Value()`. The good news is that you do not need to implement all of these
+functions of the `Context` interface – you just need to declare and modify a `Context`
+variable using functions such as `context.WithCancel()`, `context.WithDeadline()` and
+`context.WithTimeout()`.
+
+The source code of the `context` package includes the following information:
+
+{{< note >}}
+// The WithCancel, WithDeadline, and WithTimeout functions take a
+// Context (the parent) and return a derived Context (the child) and a
+// CancelFunc. Calling the CancelFunc cancels the child and its
+// children, removes the parent's reference to the child, and stops
+// any associated timers. Failing to call the CancelFunc leaks the
+// child and its children until the parent is canceled or the timer
+// fires. The go vet tool checks that CancelFuncs are used on all
+// control-flow paths.
+{{< /note >}}
+
+## A simple example
+
+The first code example will be relatively simple and illustrate the use of the
+`context.Context` type with the help of `simple.go`.
+
+### Explaining the Go code of the Example
+
+The Go code of `simple.go` is as follows:
+
+{{< file "./simple.go" go >}}
+package main
+
+import (
+        "context"
+        "fmt"
+        "os"
+        "strconv"
+        "time"
+)
+
+func f1(t int) {
+        c1 := context.Background()
+        c1, cancel := context.WithCancel(c1)
+        defer cancel()
+
+        go func() {
```

Comment:
Same as above re: documentation. A high-level explanation of what `f1` does in the code sample might be nice.

Reply:
Done.